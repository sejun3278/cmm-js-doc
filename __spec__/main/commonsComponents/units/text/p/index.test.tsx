import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import _PText from "src/main/commonsComponents/units/text/p";
import { useRouter } from "next/router";

// 가짜 router 만들기
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
const push = jest.fn();
(useRouter as jest.Mock).mockImplementation(() => ({
  push,
}));

export default describe("Text P tag Units Page Jest", () => {
  // P 태그 스냅샷
  test("Text P Tag Units Page - Sanpshot", () => {
    const { container } = render(
      <_PText
        text="p 태그 테스트"
        className="_jest_p_text_"
        styles={{ color: "red" }}
      />
    );
    // @ts-ignore
    expect(container).toMatchSnapshot();
  });

  // P 태그 존재여부 검증
  test("Text P Tag Units Page - Check has P Tag", () => {
    const { container } = render(
      <_PText
        text="p 태그 테스트"
        className="_jest_p_text_"
        styles={{ color: "red" }}
      />
    );
    const pEle = container.querySelector("p");
    expect(pEle).toBeInTheDocument(); // P 태그가 있는지 검증

    if (pEle) {
      const checkText = screen.getByText("p 태그 테스트");
      expect(checkText).toBeInTheDocument(); // 텍스트가 존재하는지 검증

      expect(pEle).toHaveClass("_jest_p_text_"); // 클래스가 있는지 검증
      expect(pEle).toHaveStyle("color: red"); // 스타일 일치하는지 검증
    }
  });
});
