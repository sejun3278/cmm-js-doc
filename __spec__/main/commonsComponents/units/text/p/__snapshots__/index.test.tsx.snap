// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text P tag Units Page Jest Text P Tag Units Page - Sanpshot 1`] = `
<div>
  <p
    class="_p_ _jest_p_text_ css-pzlb5d"
    style="color: red;"
  >
    p 태그 테스트
  </p>
</div>
`;
